### ISSUE TITLE
Add functionality explanation to the `timeout(...)` method on Streams page

### ISSUE THREAD

**Issue by SandPod** [dart-lang/site-www#5404]
### Page URL

https://dart.dev/libraries/async/using-streams

### Describe the problem

The tutorial for asynchronous streams is missing documentation about the `timeout(...)` method available on Stream objects. It is listed together with `handleError(...)` and `transform(....)` but misses a section describing its functionality.

### Expected fix

A matching description about the `timeout(...)` method describing what it is for and how it is used.

### I would like to fix this problem.

- [ ] I will try and fix this problem on dart.dev.

**Comment by huycozy** [comment]
Hi @SandPod 
Could you check https://api.flutter.dev/flutter/dart-async/Stream/timeout.html to see if it has information that you are looking for?

**Comment by SandPod** [comment]
@huycozy Yes, that documentation was good enough for my use case. Thank you 👍

The information should still probably either be summarized or linked from tutorial page.
